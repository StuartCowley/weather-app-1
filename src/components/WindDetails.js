import "../styles/WindDetails.css";
import React from "react";
import PropTypes from "prop-types";
import arrow from "../styles/img/arrow.svg";

const WindDetails = ({ speed, direction }) => {
  let rotation = "";
  if (direction === "n") {
    rotation = 0;
  }
  if (direction === "nne") {
    rotation = 23;
  }
  if (direction === "ne") {
    rotation = 45;
  }
  if (direction === "ene") {
    rotation = 68;
  }
  if (direction === "e") {
    rotation = 90;
  }
  if (direction === "ese") {
    rotation = 113;
  }
  if (direction === "se") {
    rotation = 135;
  }
  if (direction === "sse") {
    rotation = 158;
  }
  if (direction === "s") {
    rotation = 180;
  }
  if (direction === "ssw") {
    rotation = 203;
  }
  if (direction === "sw") {
    rotation = 225;
  }
  if (direction === "wsw") {
    rotation = 248;
  }
  if (direction === "w") {
    rotation = 270;
  }
  if (direction === "wnw") {
    rotation = 293;
  }
  if (direction === "nw") {
    rotation = 315;
  }
  if (direction === "nnw") {
    rotation = 338;
  }
  const rotationStyle = `rotate(${rotation}deg)`;
  const speedString = `${speed}ms`;
  return (
    <div className="wind-details">
      {speedString}
      <img alt="img" src={arrow} style={{ transform: rotationStyle }} />
    </div>
  );
};
WindDetails.propTypes = {
  speed: PropTypes.number.isRequired,
  direction: PropTypes.string.isRequired,
};

export default WindDetails;
