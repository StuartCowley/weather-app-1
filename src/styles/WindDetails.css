
.wind-details{
  display: flex;
  padding-left: 15%;
}