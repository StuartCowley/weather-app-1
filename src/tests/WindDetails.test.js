import React from "react";
import { render } from "@testing-library/react";
import WindDetails from "../components/WindDetails";

describe("WindDetails", () => {
  const validProps = {
    speed: 5,
    direction: "ne",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <WindDetails speed={validProps.speed} direction={validProps.direction} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByText, getByTestId } = render(
      <WindDetails speed={validProps.speed} direction={validProps.direction} />
    );

    expect(getByText("5ms")).toHaveClass("wind-details");
    expect(getByTestId("wind-details-image")).toHaveClass("wind-details__img");
  });
});
