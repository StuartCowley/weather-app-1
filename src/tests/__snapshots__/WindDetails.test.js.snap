// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="wind-details"
  >
    5ms
    <img
      alt="img"
      class="wind-details__img"
      data-testid="wind-details-image"
      src="arrow.svg"
      style="transform: rotate(45deg);"
    />
  </div>
</DocumentFragment>
`;
